homeassistant/components/tado/climate.py

```diff
@@ -465,11 +472,19 @@ def set_swing_mode(self, swing_mode):
     def _async_update_zone_data(self):
         """Load tado data into zone."""
         self._tado_zone_data = self._tado.data["zone"][self.zone_id]
+        self._update_offset()
         self._current_tado_fan_speed = self._tado_zone_data.current_fan_speed
         self._current_tado_hvac_mode = self._tado_zone_data.current_hvac_mode
         self._current_tado_hvac_action = self._tado_zone_data.current_hvac_action
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
+    def _update_offset(self):
+        # Add 'offset' to the key so its named correctly as an attribute in HA
+        for key in self._tado.data["device"][self._device_id]["TEMP_OFFSET"]:
+            self._tado_zone_temp_offset["offset_" + key] = self._tado.data["device"][
```

Comment:
Please set the state attribute keys explicitly. We want to see what they are and their formatting. We can use a constant dict to map between state attribute keys and device data keys.

Reply:
@MartinHjelmare thank you
Sorry just to check I follow... You mean instead of looping through the dict received to set whatever is in there, actually map each attribute? so:

offset_celsius = self._tado.data["device"][self._device_id]["TEMP_OFFSET"]["celsius"]

but with stick them in a dict map const to loop through? Assuming I got that right, will add 

Just trying to update the tests as the offset mock obviously isn't there, so will do that at the same time.